pkg/client/unversioned/kubelet.go

```diff
@@ -39,31 +40,20 @@ type HTTPKubeletClient struct {
 }
 
 func MakeTransport(config *KubeletConfig) (http.RoundTripper, error) {
-	cfg := &Config{TLSClientConfig: config.TLSClientConfig}
-	if config.EnableHttps {
-		hasCA := len(config.CAFile) > 0 || len(config.CAData) > 0
-		if !hasCA {
-			cfg.Insecure = true
```

Comment:
Should be in `KubletConfig.transportConfig()`. That function is supposed to create the proper transport.Config object, and that means considering EnableHttps.